ni_measurementlink_service/measurement/service.py

```diff
@@ -154,16 +161,28 @@ def __init__(
         with open(service_config_path) as service_config_file:
             service_config = json.load(service_config_file)
 
-        self.measurement_info = MeasurementInfo(
-            display_name=service_config["services"][0]["displayName"],
-            version=version,
-            ui_file_paths=ui_file_paths,
-        )
-
-        self.service_info = ServiceInfo(
-            service_class=service_config["services"][0]["serviceClass"],
-            description_url=service_config["services"][0]["descriptionUrl"],
-        )
+        service_found = False
+        for service in service_config["services"]:
+            if (
+                service_class is None
+                or service["serviceClass"] == service
+            ):
```

Comment:
`service["serviceClass"] == service` looks wrong.

Consider extracting the search into a helper method.

Consider using a more specific exception class like `RuntimeError`.

FYI, here's a way to eliminate the loop:
```
try: 
   service = next(s for s in service_config["services"] if service_class is None or service["serviceClass"] == service_class)
except StopIteration:
   raise RuntimeError(...)
```

Reply:
Oops. Yeah that is meant to be `service_class`.

I think your loop elimination is going to trigger the line length analyzer.

Reply:
consider using filter https://docs.python.org/3/library/functions.html#filter

Reply:
@mshafer-NI Something like this?
```python
service = next(
    filter(
        lambda s: service_class is None or s["serviceClass"] == service_class,
        service_config["services"],
    ),
    None,
)

if not service:
    raise RuntimeError(
        f"Service class '{service_class}' not found in '{service_config_file}'"
    )

self.measurement_info = MeasurementInfo(
    display_name=service["displayName"],
    version=version,
    ui_file_paths=ui_file_paths,
)

self.service_info = ServiceInfo(
    service_class=service["serviceClass"],
    description_url=service["descriptionUrl"],
)
```

Reply:
or:
```python
services = list(
    filter(
        lambda s: service_class is None or s["serviceClass"] == service_class,
        service_config["services"],
    ),
)

if not services:
    raise RuntimeError(
        f"Service class '{service_class}' not found in '{service_config_file}'"
    )

service = services[0]
```

Reply:
or, since service_class isn't changing:
```python
if service_class is None:
   service = next(service in service_config["services"])
else:
   service = next((s for s in service_config["services"] if s["serviceClass"] == service_class), None)
   if not service :
       pass # TODO: raise

self.measurement_info = MeasurementInfo(
    display_name=service["displayName"],
    version=version,
    ui_file_paths=ui_file_paths,
)

self.service_info = ServiceInfo(
    service_class=service["serviceClass"],
    description_url=service["descriptionUrl"],
)
```

Reply:
```
if service_class is None:
   service = next(service_config["services"], None)
else:
   service = next((s for s in service_config["services"] if s["serviceClass"] == service_class), None)
if not service:
   raise RuntimeError(...)
```